lib/async_hooks.js

```diff
@@ -207,42 +207,30 @@ function triggerAsyncId() {
 // Embedder API //
 
 class AsyncResource {
-  constructor(type, triggerAsyncId) {
-    this[async_id_symbol] = ++async_uid_fields[kAsyncUidCntr];
-    // Read and reset the current kInitTriggerId so that when the constructor
-    // finishes the kInitTriggerId field is always 0.
-    if (triggerAsyncId === undefined) {
-      triggerAsyncId = initTriggerId();
-    // If a triggerAsyncId was passed, any kInitTriggerId still must be null'd.
-    } else {
-      async_uid_fields[kInitTriggerId] = 0;
-    }
-    this[trigger_id_symbol] = triggerAsyncId;
-
-    if (typeof type !== 'string' || type.length <= 0)
-      throw new TypeError('type must be a string with length > 0');
+  constructor(type, triggerAsyncId = initTriggerId()) {
```

Comment:
So just an FYI since this effectively changes the signature of the ctor (now `AsyncResource.length == 1` it would have been `semver-major`, but since this congruent with [the docs](https://nodejs.org/api/async_hooks.html#async_hooks_asyncresource_type_triggerid), I'd call this a bug fix.